File: packages/sentinel-app/lib/shared/models/activity_item.dart
/// Activity events from Claude Code (PostToolUse, Notification, Stop, etc.)
enum ActivityType {
  toolCompleted('tool_completed'),
  notification('notification'),
  stop('stop'),
  taskCompleted('task_completed'),
  sessionEnded('session_ended'),
  userMessage('user_message'),
  claudeResponse('claude_response'),
  claudeStatus('claude_status'),
  terminal('terminal');

  final String value;
  const ActivityType(this.value);

  static ActivityType fromString(String s) {
    return ActivityType.values.firstWhere(
      (e) => e.value == s,
      orElse: () => ActivityType.toolCompleted,
    );
  }
}

class ActivityItem {
  final String id;
  final ActivityType type;
  final String summary;
  final String? toolName;
  final DateTime timestamp;
  final String? stopReason;
  final String? message;

  ActivityItem({
    required this.id,
    required this.type,
    required this.summary,
    this.toolName,
    required this.timestamp,
    this.stopReason,
    this.message,
  });

  factory ActivityItem.fromJson(Map<String, dynamic> json) {
    return ActivityItem(
      id: json['id'] as String? ?? DateTime.now().millisecondsSinceEpoch.toString(),
      type: ActivityType.fromString(json['type'] as String? ?? ''),
      summary: json['summary'] as String? ?? json['message'] as String? ?? '',
      toolName: json['toolName'] as String?,
      timestamp: DateTime.now(),
      stopReason: json['stopReason'] as String?,
      message: json['message'] as String?,
    );
  }

  bool get isError => stopReason == 'error';
}

/// Terminal output line
class TerminalLine {
  final String id;
  final String text;
  final DateTime timestamp;

  TerminalLine({required this.id, required this.text, required this.timestamp});
}

File: Sentinel/Models/ActivityItem.swift
import Foundation

enum ActivityType: String, Codable {
    case toolCompleted  = "tool_completed"
    case notification   = "notification"
    case stop           = "stop"
    case taskCompleted  = "task_completed"
    case sessionEnded   = "session_ended"
    case userMessage    = "user_message"
    case claudeResponse = "claude_response"
    case claudeStatus   = "claude_status"

    var systemImage: String {
        switch self {
        case .toolCompleted:  "checkmark.circle"
        case .notification:   "bell"
        case .stop:           "flag.checkered"
        case .taskCompleted:  "checkmark.seal"
        case .sessionEnded:   "clock"
        case .userMessage:    "text.bubble"
        case .claudeResponse: "bubble.left.fill"
        case .claudeStatus:   "ellipsis.bubble"
        }
    }

    var label: String {
        switch self {
        case .toolCompleted:  String(localized: "工具完成")
        case .notification:   String(localized: "通知")
        case .stop:           String(localized: "任务结束")
        case .taskCompleted:  String(localized: "子任务完成")
        case .sessionEnded:   String(localized: "会话结束")
        case .userMessage:    String(localized: "用户消息")
        case .claudeResponse: String(localized: "Claude 回复")
        case .claudeStatus:   String(localized: "思考中...")
        }
    }
}

struct ActivityItem: Identifiable, Codable {
    let id: String
    let type: ActivityType
    let summary: String
    let toolName: String?
    let timestamp: Date
    let stopReason: String?
    let message: String?

    var isError: Bool {
        stopReason == "error"
    }
}
